src/control/server/ctl_system.go

```diff
@@ -200,51 +200,96 @@ func (svc *ControlService) SystemStop(ctx context.Context, req *ctlpb.SystemStop
 	return resp, nil
 }
 
-// restartHarness restarts ranks managed by harness running on remote host.
-// TODO: specify specific rank(s) to restart in request
-func (svc *ControlService) restartHarness(ctx context.Context, leader *IOServerInstance, addr string) error {
-	_, err := leader.msClient.Restart(ctx, addr, &mgmtpb.RestartRanksReq{})
+// startRemoteHarness starts ranks managed by harness running on remote host.
+//
+// TODO: specify specific rank(s) to start in request
+func (svc *ControlService) startRemoteHarness(ctx context.Context, leader *IOServerInstance, addr string, ranks ...uint32) error {
+	req := mgmtpb.StartRanksReq{}
+	req.Ranks = make([]uint32, 0, maxIoServers)
+
+	if len(ranks) > maxIoServers {
+		return errors.New("number of of ranks exceeds maximum")
+	}
+	if len(ranks) > 0 {
+		for _, rank := range ranks {
+			req.Ranks = append(req.Ranks, rank)
+		}
+	}
+
+	// TODO: populate response with DER codes for each rank start.
+	_, err := leader.msClient.Start(ctx, addr, &req)
 
 	return err
 }
 
-// restartMembers sends multicast RestartRanks gRPC requests to host addresses in
+func resolveLeaderMemberAddr(leader *IOServerInstance, members system.Members) (string, error) {
+	var leaderMember *system.Member
+
+	msAddr, err := leader.msClient.LeaderAddress()
+	if err != nil {
+		return "", err
+	}
+
+	aps, err := resolveAccessPoints([]string{msAddr})
+	if err != nil {
+		return "", err
+	}
+
+	for _, member := range members {
+		if member.Addr.String() == aps[0].String() {
+			leaderMember = member
+		}
+	}
+	if leaderMember == nil {
+		return "", errors.Errorf("MS leader address %s not found in membership",
+			aps[0].String())
+	}
+
+	return leaderMember.Addr.String(), nil
+}
+
+// startMembers sends multicast StartRanks gRPC requests to host addresses in
 // system membership list. Each host address represents a gRPC server associated
 // with a harness managing one or more data-plane instances (DAOS system members).
-func (svc *ControlService) restartMembers(ctx context.Context, leader *IOServerInstance) (map[string]error, error) {
+//
+// TODO: specify the ranks managed by the harness that should be started.
+func (svc *ControlService) startMembers(ctx context.Context, leader *IOServerInstance) (map[string]error, error) {
 	members := svc.membership.Members()
 	results := make(map[string]error)
 
-	msAddr, err := leader.msClient.LeaderAddress()
+	leaderAddr, err := resolveLeaderMemberAddr(leader, members)
```

Comment:
I don't understand what problem this is solving. Is this meant to try and future-proof the code for when we have multiple MS replicas?

Reply:
this resolves the hostname to IP address so we can properly match when iterating through the membership addresses, leaderAddress will otherwise return a hostname from the config which might not be an IP address and therefore might not match the address in the membership. membership addresses have been resolved and will always be an IP format.

Reply:
Ok, thanks for the explanation. In the long run, it seems like it would be better to have the LeaderAddress() method return a resolved address. That way all users of the address can be consistent without developers having to remember to run it through another function. We can address that in a follow up, though.

Reply:
agreed